### ISSUE TITLE
osc fails to check out an empty project as project

### ISSUE THREAD

**Issue by frispete** [openSUSE/osc#858]
Hi,

after creating an empty project `home:frispete:tmp` on OBS, checking it out results in a package like directory, although it should constitute a project, shouldn't it?

```
$ osc co home:frispete:tmp
A    home:frispete:tmp
At revision None.
$ cd home:frispete:tmp
$ osc mkpac test
'.../home:frispete:tmp' is not a working copy
$ l .osc
insgesamt 24
drwxr-xr-x 2 hp lisa 131  3. Nov 18:52 ./
drwxr-xr-x 3 hp lisa  26  3. Nov 18:52 ../
-rw-r--r-- 1 hp lisa  25  3. Nov 18:52 _apiurl
-rw-r--r-- 1 hp lisa  24  3. Nov 18:52 _files
-rw-r--r-- 1 hp lisa 123  3. Nov 18:52 _meta
-rw-r--r-- 1 hp lisa   4  3. Nov 18:52 _osclib_version
-rw-r--r-- 1 hp lisa  18  3. Nov 18:52 _package
-rw-r--r-- 1 hp lisa   1  3. Nov 18:52 _project
```

**Comment by marcus-h** [comment]
On 2020-11-03 10:13:04 -0800, Hans-Peter Jansen wrote:
> after creating an empty project `home:frispete:tmp` on OBS, checking it out results in a package like directory, although it should constitute a project, shouldn't it?
>
Hmm did a home:frispete:tmp dir/package wc exist before?

> ```
> $ osc co home:frispete:tmp
> A    home:frispete:tmp
> At revision None.
> $ cd home:frispete:tmp
> $ osc mkpac test
> '.../home:frispete:tmp' is not a working copy
> $ l .osc
> insgesamt 24
> drwxr-xr-x 2 hp lisa 131  3. Nov 18:52 ./
> drwxr-xr-x 3 hp lisa  26  3. Nov 18:52 ../
> -rw-r--r-- 1 hp lisa  25  3. Nov 18:52 _apiurl
> -rw-r--r-- 1 hp lisa  24  3. Nov 18:52 _files
> -rw-r--r-- 1 hp lisa 123  3. Nov 18:52 _meta
> -rw-r--r-- 1 hp lisa   4  3. Nov 18:52 _osclib_version
> -rw-r--r-- 1 hp lisa  18  3. Nov 18:52 _package
> -rw-r--r-- 1 hp lisa   1  3. Nov 18:52 _project
> ```
Works for me

marcus@linux:~> rm -rf home\:frispete\:tmp/
marcus@linux:~> osc co home:frispete:tmp
A    home:frispete:tmp
marcus@linux:~> cd home\:frispete\:tmp/
marcus@linux:~/home:frispete:tmp> osc mkpac foo
A    foo
marcus@linux:~/home:frispete:tmp> find .
.
./.osc
./.osc/_project
./.osc/_apiurl
./.osc/_packages
./foo
./foo/.osc
./foo/.osc/_project
./foo/.osc/_package
./foo/.osc/_apiurl
./foo/.osc/_files
./foo/.osc/_osclib_version
marcus@linux:~/home:frispete:tmp>

**Comment by frispete** [comment]
not here:

```
$ rm -r home\:frispete\:tmp
$ osc co home\:frispete\:tmp
A    home:frispete:tmp
At revision None.
$ cd home\:frispete\:tmp
$ osc mkpac foo
'.../home:frispete:tmp' is not a working copy
$ osc --version
0.170.0
$ alias osc 
alias osc='osc -A https://api.opensuse.org'
```

Any idea, which configuration setting could affect this?

**Comment by frispete** [comment]
The `At revision None.` looks suspicious. What's probably slightly unusual, my local obs repos are hosted on a nfs4 fs (15.2 server). From a cursory look, I don't have anything unusual in my config.

**Comment by marcus-h** [comment]
On 2020-11-04 01:17:45 -0800, Hans-Peter Jansen wrote:
> not here:
>
> ```
> $ rm -r home\:frispete\:tmp
> $ osc co home\:frispete\:tmp
> A    home:frispete:tmp
> At revision None.
> $ cd home\:frispete\:tmp
> $ osc mkpac foo
> '.../home:frispete:tmp' is not a working copy

Is "..." really part of the output or did you just manually
shortened the path?

Does a "$PWD/.osc" dir exist?

**Comment by frispete** [comment]
> Is "..." really part of the output or did you just manually shortened the path?

Manually shortened the slightly longish path.

> Does a "$PWD/.osc" dir exist?

Bingo, thanks for the hint. For some reason, I had a .osc package folder in the root of my obs projects:

```
 .osc
insgesamt 32
drwxrwxr-x   2 hp lisa  115  4. Nov 10:13 ./
drwxr-xr-x 139 hp lisa 8192  4. Nov 10:13 ../
-rw-rw-r--   1 hp lisa   25 17. Feb 2014  _apiurl
-rw-rw-r--   1 hp lisa    4 18. Apr 2018  _commit_msg
-rw-r--r--   1 hp lisa 1244  4. Nov 10:13 _packages
-rw-rw-r--   1 hp lisa    1 17. Feb 2014  _project
```

And it's "_packages" piled up ~20 projects, which seems to resemble the occurrences of this issue over time.

Finally fixed. Thank you.

**Comment by marcus-h** [comment]
On 2020-11-04 15:15:52 +0000, Hans-Peter Jansen wrote:
> > Is "..." really part of the output or did you just manually shortened the path?
>
> Manually shortened the slightly longish path.
>
> > Does a "$PWD/.osc" dir exist?
>
> Bingo, thanks for the hint. For some reason, I had a .osc package folder in the root of my obs projects:
>
> ```
>  .osc
> insgesamt 32
> drwxrwxr-x   2 hp lisa  115  4. Nov 10:13 ./
> drwxr-xr-x 139 hp lisa 8192  4. Nov 10:13 ../
> -rw-rw-r--   1 hp lisa   25 17. Feb 2014  _apiurl
> -rw-rw-r--   1 hp lisa    4 18. Apr 2018  _commit_msg
> -rw-r--r--   1 hp lisa 1244  4. Nov 10:13 _packages
> -rw-rw-r--   1 hp lisa    1 17. Feb 2014  _project
> ```
>
That's a project working copy. What's the content of the .osc/_project
file? It must be something "strange" because it "tricks" osc so that
it assumes that <CONTENT of .osc/_project>/home:frispete:tmp is an
existing package.

**Comment by frispete** [comment]
> That's a project working copy.

In the root of all OBS projects, there shouldn't be one.

> What's the content of the .osc/_project file?

_project just contained a newline (length: 1). 

I just removed .osc at that level altogether. 

I guess, this .osc was created some 6,5 years ago from a silly osc usage on my side. Since then, it drove its dreadful state of affairs mostly uncovered... I got, what it deserved: termination.

While at it, is there a way to create an offline project with osc?

**Comment by marcus-h** [comment]
On 2020-11-04 16:15:36 +0000, Hans-Peter Jansen wrote:
> > That's a project working copy.
>
> In the root of all OBS projects, there shouldn't be one.
>
Yes:) (I just wanted to point out that it is a project wc instead
of a package wc).

> > What's the content of the .osc/_project file?
>
> _project just contained a newline (length: 1).
>
Ah ok!

> While at it, is there a way to create an offline project with osc?
>
You can initialize an "offline" project via

cd some_dir
osc init name_of_the_project

but a subsequent osc commit in the project or one of its packages
will _not_ create the project (you could create it before the commit
via "osc meta prj name_of_the_project -e" (where you can also specify
the repos etc.)).

**Comment by frispete** [comment]
Thank you, Marcus, for your kind support. It's a real pleasure.

For the audience, `osc init name_of_the_project` will trigger a server error, that should be ignored.

**Comment by marcus-h** [comment]
On 2020-11-05 11:52:01 +0000, Hans-Peter Jansen wrote:
> For the audience, `osc init name_of_the_project` will trigger a server error, that should be ignored.
>
Oops. Fixed in PR #859 ("Do not retrieve the package list when
initializing a project via osc init").

**Comment by frispete** [comment]
You render me speechless. :1st_place_medal:
